app/src/main/proto/vss.proto

```diff
@@ -98,6 +98,64 @@ message PutObjectRequest {
 message PutObjectResponse {
 }
 
+message GetKeysSummaryRequest {
+
+  // store_id is a keyspace identifier.
+  // Ref: https://en.wikipedia.org/wiki/Keyspace_(distributed_data_store)
+  // All APIs operate within a single store_id.
+  // It is up to clients to use single or multiple stores for their use-case.
+  // This can be used for client-isolation/ rate-limiting / throttling on the server-side.
+  // Authorization and billing can also be performed at the store_id level.
+  string store_id = 1;
+
+  // A key_prefix is a string of characters at the beginning of the key. Prefixes can be used as
+  // a way to organize key-values in a similar way to directories.
+  //
+  // If key_prefix is specified, the response results will be limited to those keys that begin with
+  // the specified prefix.
+  //
+  // If no key_prefix is specified or it is empty, all the keys are eligible to be returned in the
+  // response.
+  optional string key_prefix = 2;
+
+  // last_evaluated_key is a pagination token.
+  //
+  // To query for the first page of GetKeysSummary, last_evaluated_key must not be specified.
+  //
+  // For subsequent pages, use the value that was returned as `last_evaluated_key` in the previous
+  // page's GetKeysSummaryResponse.
+  // It will be used as exclusive_start_key to perform the new query and only keys greater than
+  // the specified last_evaluated_key will be returned in the response.
+  optional string last_evaluated_key = 3;
```

Comment:
consider naming this be `pagination_token` or similar, to discourage the client from assuming anything about the nature of this field.  this will give us flexibility in the future to paginate in any order.

could probably drop the paragraph with `exclusive_start_key` (it is not defined, and the client doesn't really care)

Reply:
this makes sense, 
earlier version of this api was inspired by s3/ddb api. 
but this approach is more generic and some of the new aws api's use a similar approach along with google cloud api's.
that best practices guide was also very helpful.